x-pack/plugins/monitoring_collection/server/plugin.ts

```diff
@@ -85,6 +111,55 @@ export class MonitoringCollectionPlugin implements Plugin<MonitoringCollectionSe
     };
   }
 
+  private configureOpentelemetryMetrics(
+    serviceName?: string,
+    serviceInstanceId?: string,
+    serviceVersion?: string
+  ) {
+    const meterProvider = new MeterProvider({
+      resource: new Resource({
+        [SemanticResourceAttributes.SERVICE_NAME]: serviceName,
+        [SemanticResourceAttributes.SERVICE_INSTANCE_ID]: serviceInstanceId,
+        [SemanticResourceAttributes.SERVICE_VERSION]: serviceVersion,
+      }),
+    });
+
+    metrics.setGlobalMeterProvider(meterProvider);
+
+    const otlpConfig = this.config.opentelemetry?.metrics.otlp;
+    if (otlpConfig?.url) {
+      const url = otlpConfig.url;
+      this.logger.debug(`Registering OpenTelemetry metrics exporter to ${url}`);
+
+      const credentials = url.startsWith('https://')
+        ? grpc.credentials.createSsl()
+        : grpc.credentials.createInsecure();
```

Comment:
Pretty sure with 0.30 we can drop this (and the grpc direct dependency)

Reply:
I opened https://github.com/crespocarlos/kibana/pull/1 with the proposed change - still need to test it with/without SSL

Reply:
Sadly looks like the grpc dependency is still required to set headers, but at least we don't have to do our own secure/insecure check anymore.